pkg/framework/context/context.go

```diff
@@ -130,14 +136,25 @@ func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
 		// find generic inbound services registered for given message header
 		for _, svc := range p.msgSvcProvider.Services() {
 			h := &service.Header{}
-			err = msg.Decode(h)
 
+			err = msg.Decode(h)
 			if err != nil {
 				return err
 			}
 
 			if svc.Accept(h) {
+				var opts []interface{}
+				// TODO: create an interface in some common package
+				if svcOpts, ok := svc.(interface{ MessengerOpts() []interface{} }); ok {
```

Comment:
The interface should be explicitly defined & named rather than inline.

Reply:
Done!